.circleci/config.yml

```diff
@@ -1,68 +1,84 @@
-post_checkout: &post_checkout
-  run:
-    name: Checkout submodule and get changed files
-    command: |
-        git submodule sync
-        git submodule update --init --recursive
-        if [[ $CIRCLE_PULL_REQUEST ]]; then
-          BASE_COMMIT=$(curl -s -H "Authorization: token a61ecb2fc5b72da54431""1b3db3875c96854958a8" \
-            https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER \
-            | jq -r ".base.sha | select(. != null)")...
-        else
-          BASE_COMMIT=HEAD^
-        fi
-        if [[ $BASE_COMMIT ]]; then
-          CHANGED=$(git diff --name-only $BASE_COMMIT)
-          echo "$CHANGED"
-          echo "export CHANGED=\"$CHANGED\"" >> $BASH_ENV
-        fi
+version: 2.1
 
-node_modules_cache_key: &node_modules_cache_key
-  yarn-deps-v2-{{ checksum "yarn.lock" }}
-
-restore_node_modules_cache: &restore_node_modules_cache
-  restore_cache:
-    keys:
-      - *node_modules_cache_key
-      - yarn-deps-v2-
+commands:
+  checkout_repo:
+    steps:
+      - checkout
+      - run:
+          name: Checkout submodule
+          command: |
+              git submodule sync
+              git submodule update --init --recursive
+      - run:
+          name: Get changed files
+          command: |
+              if [[ $CIRCLE_PULL_REQUEST ]]; then
+                BASE_COMMIT=$(curl -s -H "Authorization: token a61ecb2fc5b72da54431""1b3db3875c96854958a8" \
+                  https://api.github.com/repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/pulls/$CIRCLE_PR_NUMBER \
+                  | jq -r ".base.sha | select(. != null)")...
+              else
+                BASE_COMMIT=HEAD^
+              fi
+              if [[ $BASE_COMMIT ]]; then
+                CHANGED=$(git diff --name-only $BASE_COMMIT)
+                echo "$CHANGED"
+                echo "export CHANGED=\"$CHANGED\"" >> $BASH_ENV
+              fi
 
-yarn_install: &yarn_install
-  run:
-    name: Install dependencies
-    command: yarn
+  skip_if_only_changed:
+    parameters:
+      filter:
+        type: string
+    steps:
+      - run:
+          name: Skip tests if only "<< parameters.filter >>" are changed
+          command: echo "$CHANGED" | grep -qvE '<< parameters.filter >>' || circleci step halt
 
-codecov: &codecov
-  run:
-    name: Upload code coverage reports to Codecov
-    command: ./node_modules/.bin/codecov
+  install_dependencies:
+    steps:
+      - restore_cache:
+          keys:
+            - yarn-deps-v2-{{ checksum "yarn.lock" }}
+            - yarn-deps-v2-
+      - run:
+          name: Install dependencies
+          # upgrade Yarn before installing dependencies, if needed
+          command: yarn check || { curl -o- -L https://yarnpkg.com/install.sh | bash && yarn; }
```

Comment:
I see.  Another option would be changing which image we're using for node.  The node 10 image contains yarn 1.9.2 https://github.com/nodejs/docker-node/blob/master/10/jessie/Dockerfile.  I guess that's a bigger change so we can upgrade yarn ourselves in the meantime.

Reply:
@kevinbarabash But we may (unwittingly) break something in development environment on Node 6, if we test it on Node 10.